### PR TITLE
Merge two sorted arrays without using extra space

### PR DESCRIPTION
**Pull request by nikitaaa23**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

### Description

C++ Program to merge two sorted arrays without using extra space O(1).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#5854

### CODE REVIEW

**Comment by nikitaaa23** [comment]
@atarax665 Kindly review!

**Comment by nikitaaa23** [comment]
@ankitaggarwal23 Please review

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Read the instructions mentioned on Gssoc discord yesterday. Do the changes as per that.

**Comment by nikitaaa23** [comment]
@ankitaggarwal23 Made the changes, please review.
Replit link: https://replit.com/@NikitaSaini/Mergewithoutextraspace#main.cpp

**Comment by nikitaaa23** [comment]
@ankitaggarwal23 Made the changes, please review.
Replit link: https://replit.com/@NikitaSaini/Mergewithoutextraspace#main.cpp

**Comment by nikitaaa23** [comment]
@ankitaggarwal23 I made the changes.
Replit link: https://replit.com/@NikitaSaini/Mergewithoutextraspace#main.cpp

File: C-Plus-Plus/sort/merge_without_extra_space.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// C++ Program to Merge two sorted arrays without using extra space
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/merge_without_extra_space.cpp]: - Add Multi-Line Comments 
- Add a beginner-friendly description at the top

File: C-Plus-Plus/sort/merge_without_extra_space.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	while(i>=0 && j<m){                         //Iterate through every element of arr1 and arr2 and compare elements of the arrays
+		if(arr2[j]<arr1[i]){
+	           swap(arr1[i], arr2[j]);
+	        }
+	           i--;
+	           j++;	    
+	}
+	sort(arr1, arr1+n);                         //sort the first array
+        sort(arr2, arr2+m);                         //sort the second array
+}
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/merge_without_extra_space.cpp]: You need to add comments to your code. Make sure the comments you add are above the line of code and not alongside it.

File: C-Plus-Plus/sort/merge_without_extra_space.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/merge_without_extra_space.cpp]: Remove extra space

File: C-Plus-Plus/sort/merge_without_extra_space.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        cout<<"Enter the elements of the first array : ";
+        for (i = 0; i < n; i++) {
+            cin >> arr1[i];
+        }
+        cout<<"Enter the elements of the second array : ";
+        for (i = 0; i < m; i++) {
+            cin >> arr2[i];
+        }
+        merge(arr1, arr2, n, m);                            //call the merge function on both the arrays
+        cout<<"Sorted Array : ";
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/merge_without_extra_space.cpp]: You need to add comments to your code. Make sure the comments you add are above the line of code and not alongside it.

File: C-Plus-Plus/sort/merge_without_extra_space.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+SAMPLE 2
+INPUT:
+Enter the size of the first array : 6
+Enter the size of the first array : 4
+Enter the elements of the first array : 1 5 9 10 15 20
+Enter the elements of the first array : 2 3 8 13
+
+OUTPUT:
+Sorted Array : 1 2 3 5 8 9 10 13 15 20
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/sort/merge_without_extra_space.cpp]: Remove extra test case

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by nikitaaa23** [comment]
@ankitaggarwal23 Made the changes, please check.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**